Gordon360/Utilities/ServerUtils.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.AspNetCore.Hosting.Server;
+using System.Linq;
+
+namespace Gordon360.Utilities
+{
+    public class ServerUtils
+    {
+        private readonly IServer _server;
+
+        public ServerUtils(IServer server)
+        {
+            _server = server;
+        }
+
+        public string GetAddress()
+        {
+            var baseUrl = _server.Features.Get<IServerAddressesFeature>()?.Addresses.FirstOrDefault(a => a.Contains("https"));
+            return baseUrl;
+        }
```

Comment:
A couple of thoughts about how to make this helper method more useful and resilient:
1. GitHub has included a warning (and Visual Studio probably shows the same warning), because `FirstOrDefault` will return `null` if it doesn't find a matching item in the collection. However, you've declared this method as returning a non-null `string`, meaning that code that calls this method won't know it has to handle the case where it returns null. To fix this, you can change the return type of the method to `string?`, which explicitly says that this method can return `null` instead of a string. That way, wherever this method is called, the calling code will be aware that it needs to handle the `null` case.
2. As Bennett mentioned at our meeting a while ago, it is probably worth handling the case where no `https` url is present. One way to do this is to take the first element in the collection if no element in the collection with `https` exists.
3. This is a minor thing (and also my fault since I sent this code snippet to you), but we probably want to check that the url **starts with** 'https', not that it **contains** 'https'.

```suggestion
        public string? GetAddress()
        {
            var addresses = _server.Features.Get<IServerAddressesFeature>()?.Addresses;
            var serverAddress = addresses?.FirstOrDefault(a => a.StartsWith("https")) ?? addresses?.FirstOrDefault();
            return serverAddress;
        }
```

Reply:
The following is another option for the serverAddress line:
```
            var baseUrl = _server.Features.Get<IServerAddressesFeature>()?.Addresses.OrderBy((a) => a.StartsWith('https') ? 0 : 1).FirstOrDefault();
```